### PR TITLE
0.16.1 release staging

### PR DESCRIPTION
**Pull request by adamgayoso**
also fixes to `prepare_query_data`

### CODE REVIEW

**Comment by adamgayoso** [comment]
@jjhong922 this is good to go, just going to update the scarches tutorial one more time

File: scvi/model/base/_archesmixin.py [hunk 1]
```diff
@@ -206,21 +206,31 @@ def prepare_query_anndata(
...
+                axis=1,
+                join="outer",
+                index_unique=None,
+                merge="unique",
+            )
+        else:
+            adata_out = adata
+
+        # also covers the case when new adata has more var names than old
+        if not var_names.equals(adata_out.var_names):
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:228]: think you can move both of these cases into the if case above

Comment (jjhong922) [reply scvi/model/base/_archesmixin.py:228]: not super necessary though

**Review by jjhong922 — APPROVED** [review]
lgtm. Might want to add tests that fail on the old arches code though

**Comment by adamgayoso** [comment]
meeseeksdev backport to 0.16.x

**Pull request merged**